src/main/java/com/google/googleidentity/oauth2/risc/RiscSendHandler.java

```diff
@@ -0,0 +1,194 @@
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.hash.Hashing;
+import com.google.googleidentity.oauth2.client.ClientDetails;
+import com.google.googleidentity.oauth2.client.ClientDetailsService;
+import com.google.googleidentity.oauth2.jwt.JwkStore;
+import com.google.googleidentity.oauth2.token.OAuth2AccessToken;
+import com.google.googleidentity.oauth2.token.OAuth2RefreshToken;
+import com.google.googleidentity.oauth2.token.OAuth2TokenService;
+import com.google.googleidentity.oauth2.util.OAuth2Constants.TokenTypes;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.util.StandardCharset;
+import io.jsonwebtoken.Jwts;
+import java.io.IOException;
+import java.security.Key;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.entity.EntityBuilder;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+/** Revoke tokens between a user and a client. Send risc if the client support it. */
+@Singleton
+public class RiscSendHandler {
+  private static final Logger log = Logger.getLogger("RiscHandler");
+  private static int MAX_RETRY_COUNT = 3;
+  private static Duration RETRY_INTERVAL_TIME = Duration.ofMinutes(10);
+  private static String WEB_URL =
+      System.getenv("WEB_URL") == null ? "localhost:8080" : System.getenv("WEB_URL");
+  private final OAuth2TokenService oauth2TokenService;
+  private final ClientDetailsService clientDetailsService;
+  private final JwkStore jwkStore;
+  private long jtiValue = 0l;
+
+  @Inject
+  public RiscSendHandler(
+      OAuth2TokenService oauth2TokenService,
+      JwkStore jwkStore,
+      ClientDetailsService clientDetailsService) {
+    this.oauth2TokenService = oauth2TokenService;
+    this.jwkStore = jwkStore;
+    this.clientDetailsService = clientDetailsService;
+  }
+
+  public void RevokeTokenWithClient(String username, String clientID) {
+
+    Optional<ClientDetails> client = clientDetailsService.getClientByID(clientID);
+
+    Preconditions.checkArgument(
+        clientDetailsService.getClientByID(clientID).isPresent(), "Client should exist");
+
+    List<OAuth2AccessToken> accessTokenList =
+        oauth2TokenService.listUserClientAccessTokens(username, clientID);
+
+    Optional<OAuth2RefreshToken> refreshToken =
+        oauth2TokenService.getUserClientRefreshToken(username, clientID);
+
+    oauth2TokenService.revokeUserClientTokens(username, clientID);
```

Comment:
I think put this in risc handler is a little bit strange, since revoke on our side is not a part of risc. So I think it's better to only handle risc related process here.